pkg/provision/provision.go

```diff
@@ -58,4 +57,26 @@ var (
 
 // ErrTemporary is return when a reservation source failed to contact the BCDB
 // user usually want to retry after getting this error
-var ErrTemporary = errors.New("network error contacting BCDB")
+type ErrTemporary struct {
+	err error
+}
+
+// NewErrTemporary wrap an error and mark it as temporary
+func NewErrTemporary(err error) error {
+	return ErrTemporary{err: err}
+}
+
+// Error implements the errors.Error interface
+func (e ErrTemporary) Error() string {
+	return e.err.Error()
+}
+
+// Is implements errors.Is interface
+func (e ErrTemporary) Is(target error) bool {
+	return target == ErrTemporary{}
```

Comment:
This won't work because that's a check on the value, not the type. which means target will only == ErrTemprary{} if and only if traget was created as NewErrTemporary(nil) 

I suggest changing that to 
```go
_, ok := target.(ErrTemporary)
return true
```